eth_utils/humanize.py

```diff
@@ -70,3 +71,26 @@ def humanize_hash(value: Hash32) -> str:
     head = value_as_hex[:DISPLAY_HASH_CHARS]
     tail = value_as_hex[-1 * DISPLAY_HASH_CHARS :]
     return "{0}..{1}".format(head, tail)
+
+
+def humanize_ipfs_uri(uri: URI) -> str:
+    if not _is_ipfs_uri(uri):
+        raise TypeError("%s does not look like a valid IPFS uri." % uri)
+
+    parsed = parse.urlparse(uri)
+    ipfs_hash = parsed.netloc
+    head = ipfs_hash[:DISPLAY_HASH_CHARS]
+    tail = ipfs_hash[-1 * DISPLAY_HASH_CHARS :]
+    return "ipfs://{0}..{1}".format(head, tail)
+
+
+def _is_ipfs_uri(uri: str) -> bool:
```

Comment:
In theory this should take `Any` and there should be a type check of `isinstance(value, str)` at the beginning since the `is_thing` APIs should *in theory* support passing any arbitrary type or value into them.  That also implies that the variable name should be `value` to make it more generic.

Since this isn't a public API (yet) this isn't a big deal, but I could see that changing at some point in the near-ish-term future.